src/transformers/models/bark/processing_bark.py

```diff
@@ -0,0 +1,221 @@
+
+    @classmethod
+    def from_pretrained(
+        cls, pretrained_processor_name_or_path, speaker_embeddings_file_name="speaker_embeddings.npz", **kwargs
+    ):
+        r"""
+        Instantiate a Bark processor associated with a pretrained model.
+
+        Args:
+            pretrained_model_name_or_path (`str` or `os.PathLike`):
+                This can be either:
+
+                - a string, the *model id* of a pretrained [`BarkProcessor`] hosted inside a model repo on
+                  huggingface.co. Valid model ids can be located at the root-level, like `bert-base-uncased`, or
+                  namespaced under a user or organization name, like `dbmdz/bert-base-german-cased`.
+                - a path to a *directory* containing a processor saved using the [`~BarkProcessor.save_pretrained`]
+                  method, e.g., `./my_model_directory/`.
+            speaker_embeddings_file_name (`str`, *optional*, defaults to `"speaker_embeddings.npz"`):
+                The name of the `.npz` file containing the speaker_embeddings (e.g `"speaker_embeddings.npz"`) located
+                in `pretrained_model_name_or_path`. If `None`, no speaker_embeddings is loaded.
+            **kwargs
+                Additional keyword arguments passed along to both
+                [`~tokenization_utils_base.PreTrainedTokenizer.from_pretrained`].
+        """
+
+        if speaker_embeddings_file_name is not None:
+            speaker_embeddings = get_file_from_repo(
+                pretrained_processor_name_or_path,
+                speaker_embeddings_file_name,
+                subfolder=kwargs.pop("subfolder", None),
+                cache_dir=kwargs.pop("cache_dir", None),
+                force_download=kwargs.pop("force_download", False),
+                proxies=kwargs.pop("proxies", None),
+                resume_download=kwargs.pop("resume_download", False),
+                local_files_only=kwargs.pop("local_files_only", False),
+                use_auth_token=kwargs.pop("use_auth_token", None),
+                revision=kwargs.pop("revision", None),
+            )
+            if speaker_embeddings is None:
+                warnings.warn(
+                    f"`{os.path.join(pretrained_processor_name_or_path,speaker_embeddings_file_name)}` does not exists, no preloaded speaker embeddings will be used - Make sure to provide a correct path if wanted, otherwise set `speaker_embeddings_file_name=None`."
+                )
+            else:
+                # TODO: not sure this is the safest way to load/save speaker_embeddings
+                speaker_embeddings = np.load(speaker_embeddings, allow_pickle=True)
+                
+                if len(speaker_embeddings.keys()) != 1:
+                    raise ValueError(f"`speaker_embeddings` doesn't follow the required format - uses a speaker_embeddings file saved via `np.savez(speaker_embeddings_file_name, speaker_embeddings)`.")
```

Comment:
Don't fully follow the second half of the error message - maybe we can copy what you've written in the docstrings:
```suggestion
                    raise ValueError(f"`speaker_embeddings` doesn't follow the required format - ensure that speaker embeddings are saved in a nested format, where the first level contains voice preset names (e.g `en_speaker_4`), and the second level contains `semantic_prompt`, `coarse_prompt` and `fine_prompt` embeddings, then save the speaker embeddings via `np.savez('speaker_embeddings.npz', speaker_embeddings)`.")
```